x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -1249,60 +1255,137 @@ public String getFeatureDescription() {
         return "Provides anomaly detection and forecasting functionality";
     }
 
-    @Override public void cleanUpFeature(
+    @Override
+    public void cleanUpFeature(
         ClusterService clusterService,
         Client client,
-        ActionListener<ResetFeatureStateResponse.ResetFeatureStateStatus> listener) {
+        ActionListener<ResetFeatureStateResponse.ResetFeatureStateStatus> finalListener) {
+        logger.info("Starting machine learning cleanup");
+
+        ActionListener<ResetFeatureStateResponse.ResetFeatureStateStatus> unsetResetModeListener = ActionListener.wrap(
+            success -> client.execute(SetResetModeAction.INSTANCE, SetResetModeAction.Request.disabled(), ActionListener.wrap(
+                resetSuccess -> finalListener.onResponse(success),
+                resetFailure -> {
+                    logger.warn("failed to disable reset mode after state clean up success", resetFailure);
+                    finalListener.onResponse(success);
```

Comment:
I think this should count as a failure, as it leaves ML in a sub-optimal state - there will be no notifications and no nightly cleanup.  So we'd want the user to try the reset again to have another go at removing the reset-in-progress flag.